troykahat/gpio_expander.py

```diff
@@ -0,0 +1,335 @@
+    Use analog_io to create an object.
+    """
+    def __init__(self, i2c_address):
+        """The constructor for GpioExpander class.
+
+
+        Optional parameters:
+        --------------------
+        i2c_address: int
+            Device address on the I²C bus. When called without a parameter,
+            the default address is 42.
+        """
+        wiringpi.wiringPiSetup()
+        self._i2c = wiringpi.I2C()
+        self._io = self._i2c.setupInterface("/dev/i2c-1", i2c_address)
+
+    def pinMode(self, pin, mode):
+        """Configures the pin operation mode.
+
+        Parameters:
+        -----------
+        pin: int
+            The number of the used pin, which is marked on the Troyka HAT board
+            with an index from 0 to 7 on the group "Analog IO".
+        mode: Mode
+            Pin operation mode. See the available modes in the Mode class.
+
+        Returns:
+        --------
+        None
+        """
+        data = self._reverse_uint16(1 << pin)
+        if mode == Mode.INPUT:
+            self._i2c.writeReg16(self._io, GPIO_EXPANDER_PORT_MODE_INPUT, data)
+        elif mode == Mode.INPUT_PULLUP:
+            self._i2c.writeReg16(
+                self._io, GPIO_EXPANDER_PORT_MODE_PULLUP, data)
+        elif mode == Mode.INPUT_PULLDOWN:
+            self._i2c.writeReg16(
+                self._io, GPIO_EXPANDER_PORT_MODE_PULLDOWN, data)
+        elif mode == Mode.OUTPUT:
+            self._i2c.writeReg16(
+                self._io, GPIO_EXPANDER_PORT_MODE_OUTPUT, data)
+
+    def digitalRead(self, pin):
+        """Reads a digital signal value from a pin of the board.
+
+        Parameters:
+        -----------
+        pin: int
+            The number of the used pin, which is marked on the Troyka HAT board
+            with an index from 0 to 7 on the group "Analog IO".
+
+        Returns:
+        --------
+        True: High level, which conforming to 3.3V.
+        False: Low level, which conforming to 0V.
+        """
+        mask = 1 << pin
+        result = 0
+        if self._digitalReadPort() & mask:
+            result = 1
+        return result
+
+    def digitalWrite(self, pin, value):
+        """
+        Sets the digital signal on the pin of the board.
+
+        Parameters:
+        -----------
+        pin: int
+            The number of the used pin, which is marked on the Troyka HAT board
+            with an index from 0 to 7 on the group "Analog IO".
+        value: bool
+            True / HIGH / 1 — setting high level.
+            False / LOW / 0 — setting low level.
+
+        Returns:
+        --------
+        None
+        """
+        data = self._reverse_uint16(1 << pin)
+        if value:
+            self._i2c.writeReg16(
+                self._io, GPIO_EXPANDER_DIGITAL_WRITE_HIGH, data)
+        else:
+            self._i2c.writeReg16(
+                self._io, GPIO_EXPANDER_DIGITAL_WRITE_LOW, data)
```

Comment:
Тут наверное можно применить принцип DRY:
```suggestion
            state = (
                GPIO_EXPANDER_DIGITAL_WRITE_HIGH if value else GPIO_EXPANDER_DIGITAL_WRITE_LOW
            )
            self._i2c.writeReg16(self._io, state, data)
```
Имена констант, кстати, очень длинные. Может имеет смысл их уменьшить?

Reply:
1. Применил принцип DRY.
2. Т.к. константы теперь в самом классе, их имена удалось сократить.